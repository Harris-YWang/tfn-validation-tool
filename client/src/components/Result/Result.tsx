import React from 'react';
import { IResultProps } from '../../types';

const Result = ({ validity }: IResultProps) =>
	validity ? <div className="alert alert-primary mt-3">This TFN is {validity}.</div> : null;

export default Result;
