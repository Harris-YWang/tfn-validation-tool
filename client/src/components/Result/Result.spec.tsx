import { shallow } from 'enzyme';
import React from 'react';
import Result from './index';

describe('Result Component', () => {
	it('should render when there is an validity value', () => {
		const validity = 'valid';
		const wrapper = shallow(<Result validity={validity} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should render nothing when no validity value', () => {
		const validity = '';
		const wrapper = shallow(<Result validity={validity} />);

		expect(wrapper.text()).toEqual('');
	});

	it('should render primary alert', () => {
		const validity = 'valid';
		const wrapper = shallow(<Result validity={validity} />);

		expect(wrapper.find('div.alert-primary').length).toBe(1);
	});
});
