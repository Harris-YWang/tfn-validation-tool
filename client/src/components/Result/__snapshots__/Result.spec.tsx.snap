// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Component should render when there is an validity value 1`] = `ShallowWrapper {}`;
